lizmap/modules/lizmap/lib/Request/Proxy.php

```diff
@@ -942,6 +975,65 @@ public static function clearLayerCache($repository, $project, $layer)
 
         return true;
     }
+
+    /**
+     * check if $body contains a '__echo__=&' param.
+     *
+     * @return bool
+     */
+    public static function hasEchoInBody(string $body)
+    {
+        $encodedEchoParam = '%5F%5Fecho%5F%5F=&';
+
+        return strstr($body, $encodedEchoParam);
+    }
+
+    /**
+     * log the URL and body in the 'echoproxy' log file
+     * add a md5 hash of the string to help retrieving it
+     * NOTE : currently we log only the url & body, thus it don't really need to be logged
+     * because the same url & body are need to retreive the content
+     * but the function will be usefull when it'll log additionnal content
+     * (HTTP-Header, content-type, ...).
```

Comment:
```suggestion
     * Log the URL and its body in the 'echoproxy' log file
     * We add a md5 hash of the string to help retrieving it later
     * NOTE : currently we log only the url & body, thus it doesn't really need to be logged
     * because the same url & body are needed to retreive the content
     * but the function will be useful when it will log additionnal content
     * (HTTP-Header, content-type, ...).
```